src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -997,6 +993,18 @@ impl_compose_tree (const char      *treefile_pathstr,
   g_hash_table_replace (self->metadata, g_strdup ("rpmostree.inputhash"),
                         g_variant_ref_sink (g_variant_new_string (new_inputhash)));
 
+  /* We might have short circuted above for e.g. --dry-run */
+  self->do_commit = TRUE;
```

Comment:
Hmm, have you considered hoisting this up to a higher level and have an `out_changed` parameter instead? Then for `install tree` do something like:

```c
if (changed)
  /* call process_touch_if_changed() */
```

And for `compose tree` something like:

```c
if (changed)
  {
    if (!opt_dry_run)
      /* call impl_commit_tree () */

    /* call process_touch_if_changed() */
  }
```

?